src/Runner.Worker/FileCommandManager.cs

```diff
@@ -182,35 +185,75 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
                     return;
                 }
 
-                if (fileSize > AttachmentSizeLimit)
-                {
-                    context.Error(String.Format(Constants.Runner.UnsupportedSummarySize, AttachmentSizeLimit / 1024, fileSize / 1024));
-                    Trace.Info($"Step Summary file ({filePath}) is too large ({fileSize} bytes); skipping attachment upload");
+                Trace.Verbose($"Step Summary file exists: {filePath} and has a file size of {fileSize} bytes");
+                if (context.Global.Variables.Get("ResultsReceiverEndpoint") != null) {
+                    Trace.Info("Using new Actions Results Job Summaries upload");
+
+                    var ResultsReceiverEndpointAddr = context.Global.Variables.Get("ResultsReceiverEndpoint");
+
+                    var token = context.Global.Variables.Get("System.AccessToken");
+                    var resultsClient = new ResultsHttpClient(token, ResultsReceiverEndpointAddr);
+                    var planId = context.Global.Variables.Get("System.PlanId");
+                    var jobId = context.Global.Variables.Get("System.JobId");
+                    var stepId = context.Global.Variables.Get("System.StepId");
+                    var stepSummaryUploadUrl = await resultsClient.GetStepSummaryUploadUrlAsync(jobId, planId, stepId, new CancellationToken());
```

Comment:
You should not make any sync IO call (ex: HTTP request) when `processCommand` as it will block STDOUT/ERR from the process and slow down the customer's workflow job, follow the existing pattern to ask the `JobServerQueue` to process these files in the background.

Reply:
@TingluoHuang This is great info, thank you! Since this will be uploading to a cloud service provider I'm guessing I should be creating a new ConcurrentQueue in the JobServerQueue to handle this type of file upload? Just wanted to verify 😄

Reply:
sure, or fork the logic when right before upload files, to log/to attachment/to your new thing.